src/components/Dropdown.tsx

```diff
@@ -0,0 +1,123 @@
+import React, {
+  useCallback,
+  useMemo,
+  useState,
+  useRef,
+  useEffect,
+} from 'react';
+import debounce from 'lodash.debounce';
+import classNames from 'classnames';
+import { Person } from '../types/Person';
+
+interface Props {
+  people: Person[];
+  delay: number;
+  onSelected?: (user: Person) => void,
+}
+
+export const Dropdown: React.FC<Props> = React.memo(({
```

Comment:
`React.memo` is a bad idea for such component

Let the end user decide if it is needed